lib/ruby_lsp/requests/formatting.rb

```diff
@@ -79,6 +94,11 @@ def formatted_file
           raise InvalidFormatter, "Unknown formatter: #{@formatter}"
         end
       end
+
+      sig { returns(T::Boolean) }
+      def syntax_tree_as_direct_dependency?
+        Bundler::LockfileParser.new(Bundler.read_file(Bundler.default_lockfile)).dependencies.key?("syntax_tree")
```

Comment:
We should perform IO in the middle of requests or it creates the risk of switching threads in the middle of formatting. Also, it's wasteful to check this every time, we can just check this once while loading the class.